executionlayer/execution-layer.go

```diff
@@ -761,3 +764,39 @@ func (e *CachingExecutionLayer) GetRPInfo(pubkey rptypes.ValidatorPubkey) (*RPIn
 func (e *CachingExecutionLayer) REthAddress() *common.Address {
 	return e.rEth.Address
 }
+
+// ValidateEIP1271 validates an EIP-1271 signature
+func (e *CachingExecutionLayer) ValidateEIP1271(dataHash [32]byte, signature []byte, address common.Address) (bool, error) {
+	// Define the ABI for the isValidSignature function
+	const abiJSON = `[{"inputs":[{"name":"_hash","type":"bytes32"},{"name":"_signature","type":"bytes"}],"name":"isValidSignature","outputs":[{"type":"bytes4"}],"stateMutability":"view","type":"function"}]`
+
+	parsedABI, err := abi.JSON(strings.NewReader(abiJSON))
```

Comment:
nit: parsing this over and over is technically inefficient, even if the "work" is slow. It's on the "hot path", so we should consider adding a `func getEIP1271ABI() *abi.ABI` function which lazily parses it and stores it in a global pointer, returning it if it's already been parsed.

If there's an error parsing the string, we can panic, since the string is constant at runtime, so no need to return an error.

Reply:
Good callout. Will fix.

Reply:
@jshufro https://github.com/Rocket-Rescue-Node/rescue-proxy/pull/68/commits/2740bb8afd025e79cf0b30f7d768b6dea000bfd5